evap/evaluation/models.py

```diff
@@ -1044,6 +1044,14 @@ def unlogged_fields(self):
     def is_general(self):
         return self.contributor_id is None
 
+    @property
+    def can_be_deleted(self):
+        return (self.pk is None and self.is_general) or (
+            not RatingAnswerCounter.objects.filter(contribution=self).exists()
+            and not TextAnswer.objects.filter(contribution=self).exists()
+            and not self.is_general
+        )
```

Comment:
So, this is equivalent to the following, right? Why do we care about the primary key when the contribution is general but don't when it is not? I would best prefer the version below with a comment about the pk :)
```suggestion
        if self.is_general:
            return self.pk is None
        else:
            return (
                not RatingAnswerCounter.objects.filter(contribution=self).exists()
                and not TextAnswer.objects.filter(contribution=self).exists()
            )
```

Reply:
Yes it is equivalent, and much more readable 👍
If the question was not rhetorical: If a contribution is the general contribution for an evaluation, it should not be deletable at all. But the extra contribution in the Contribution formset is considered general as well, based on the property. So to distinct both of them, the pk is used because the real general contribution for an evaluation does already exist in the database.

Reply:
Radical opinion: Make the flow even more human, which the following structure:
- In general, deletion is not possible
- Exception: Object is not yet stored in database (technically no deletion, but required for the button)
- Exception: No Answers exist
which yields
```python
@property
def can_be_deleted(self):
    if self.pk is None:
        return True  # not stored in the DB. Required so temporary instances in the formset can be deleted.
    
    if not self.ratinganswercounters_set.exists() and not self.textanswers_set.exists():
        return True
    
    return False
```

using `self.ratinganswercounters_set` and the equivalent for the textanswers allows prefetching for the property, so we don't always need 2 queries to compute this property.